src/internal_telemetry/allocations/allocator/tracing_allocator.rs

```diff
@@ -96,6 +102,17 @@ unsafe impl<A: GlobalAlloc, T: Tracer> GlobalAlloc for GroupedTraceableAllocator
             },
         );
     }
+
+    #[inline]
+    unsafe fn realloc(&self, ptr: *mut u8, layout: Layout, new_size: usize) -> *mut u8 {
+        let new_layout = Layout::from_size_align_unchecked(new_size, layout.align());
+        let new_ptr = self.alloc(new_layout);
+
+        std::ptr::copy_nonoverlapping(ptr, new_ptr, std::cmp::min(layout.size(), new_size));
+        self.dealloc(ptr, layout);
+
+        new_ptr
+    }
```

Comment:
Doesn't seem like we need to be overriding this? Especially given that it's omitting the null check to make sure the new allocation succeeded, which we definitely want to keep around.

Reply:
Overriding is mainly for the benefit of inlining. We also need to decide if we want to handle nulls. Currently we panic in case of receiving a null ptr for wrapped allocations.

Reply:
But if we aren't tracking allocations, we never call into `get_wrapped_allocation`, we just call `self.allocator.alloc(...)` directly.

I think I'd rather lose the slight performance boost to keep safety invariants in place.

Reply:
I ended up removing the aborting part

Reply:
now we return null pointer if allocation failed